iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -126,21 +134,154 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
     m_memoryManagerOfCurrentProcess = m_segmentInfo.m_memoryManager;
 }
 
-void ProcessManager::killAllProcesses() noexcept
+void ProcessManager::killAllProcesses(const units::Duration finalKillTime) noexcept
 {
     std::lock_guard<std::mutex> g(m_mutex);
+    cxx::vector<bool, MAX_PROCESS_NUMBER> processStillRunning(m_processList.size(), true);
+    int i = 0;
+
+    auto yield = []() { std::this_thread::sleep_for(std::chrono::milliseconds(250)); };
 
     // send SIGTERM to all running applications
-    typename ProcessList_t::iterator it = m_processList.begin();
-    const typename ProcessList_t::iterator itEnd = m_processList.end();
+    for (auto& process : m_processList)
+    {
+        // if it was killed we need to check if it really has terminated, if we can't kill it, we consider it
+        // terminated
+        processStillRunning[i] = requestShutdownOfProcess(process, ShutdownPolicy::SIG_TERM, ShudownLog::FULL);
+        ++i;
+    }
 
-    for (; itEnd != it; ++it)
+    // wait for process to complete
+    bool allProcessFinished = false;
+    posix::Timer finalKillTimer(finalKillTime);
+
+    yield();
+    auto waitForKillProcessed = [&]() {
+        bool processStateChanged = true;
```

Comment:
This variable name is really confusing. 
Reading the could for the first time, the first question I have is "why has the process state changed before doing anything" ?

Something like `shouldCheckProcessState` would be clearer (to me at least).

Reply:
I have a not so strict rule to let all boolean start with words like is, has or like @ithier suggested should!

Reply:
Agreed.